var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -106,7 +153,31 @@ def configure_args(self):
                 args.append('--disable-llvm-shared-libs')
             args.append('--with-llvm-prefix=%s' % spec['llvm'].prefix)
 
-        if drivers:
-            args.append('--with-gallium-drivers=' + ','.join(drivers))
+        args.append('--with-gallium-drivers=' + ','.join(drivers))
+
+        # Avoid errors due to missing clock_gettime symbol:
+        arch = spec.architecture
+        if arch.platform == 'linux':
+            args.extend(['LIBS=-lrt'])
 
         return args
+
+    def configure(self, spec, prefix):
+        """Configure mesa, detecting if libsysfs is required
+        for DRI support on the build host.
+        """
+        options = ['--prefix={0}'.format(prefix)] + self.configure_args()
+        configure = inspect.getmodule(self).configure
+
+        with working_dir(self.build_directory, create=True):
+            try:
+                # First attempt uses libudev:
+                configure(*options)
+            except:
```

Comment:
It would be better to except only a specific error class. For example, what if someone presses <kbd>Ctrl</kbd>+<kbd>C</kbd>?